packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ClientMethod.java

```diff
@@ -47,6 +48,7 @@ public class ClientMethod {
     /**
      * The parameters of this ClientMethod.
      */
+    private final List<ClientMethodParameter> _parameters;
     private final List<ClientMethodParameter> parameters;
     private final List<ClientMethodParameter> methodParameters;
```

Comment:
May worth a few comments on these variation of ##parameters.
e.g. `methodParameter` would be sorted version of `parameters`.
What's the `_parameters` for?

Reply:
Good catch, I should have added the docs. So, the `_parameters` is an immutable list of ALL parameters. Now that `ClientMethod` exposes the `newBuilder()` API, it uses this immutable list to create a new builder instance. Originally (before `newBuilder`), the ClientMethod constructor was making a copy of  "parameters" constructor argument and returning it from `getParameters()`, which was mutable by the caller. This behavior has been maintained, but we can change it.

I've updated the doc to:

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/239b5b65-bf7f-4900-be52-a5c20e437fc6" />